File: submissions/StudentTwo SurnameTwo - fzampirolli3/2025-08-27-08-14-06/Q1.py
class CentralNotificacoes:
    _instancia = None
    
    def __new__(cls):
        if cls._instancia == None:
            cls._instancia = super().__new__(cls)
            cls._instancia.msg = ["Nova Conexão"]
            print("Central iniciada")
            
        else:
            print("Central já ativa")
            
        return _instancia
    def adicionar(self, msg):
        self.msg.append(msg)
        print(f"Nova notificação: {msg}")
    
    def listar(self):
        for msg in self.msg:
            print(f"{msg}")
            
msg = input()
comando = input()

app = CentralNotificacoes()

if comando == "adicionar":
    app.adicionar(msg)
    
elif comando == "listar":
    app.adicionar(msg)
    app.listar()
    
elif comando == "nova_instancia":
    app = CentralNotificacoes()

File: submissions/StudentOne SurnameOne - fzampirolli/2025-08-27-09-09-15/Q1.java
import java.io.*;
import java.util.*;

public class Q1{
    
    public static class CentralNotificacoes{
        
        // o teste 4 me parece estar com as saidas erradas, ele pede 'alerta de bateria'
        // mas o usuario nunca da input de alerta de bateria
        
        static CentralNotificacoes instance = null;
        
        ArrayList<String> notificacoes = new ArrayList<>();
        
        private CentralNotificacoes(){
            
        }
        
        public static CentralNotificacoes getInstance(){
            if(instance ==  null){
                instance = new CentralNotificacoes();
                System.out.println("Central iniciada");
            } else {
                System.out.println("Central já ativa");
            }
            return instance;
        }
        
        public void listar(){
            System.out.println("Notificações:");
            for (String notif : notificacoes){
                System.out.println(notif);
            }
        }
        
        public void adicionar(String notif){
            System.out.println("Nova notificação: " + notif);
            notificacoes.add(notif);
        }
    }
    
    public static void main(String args[]) throws IOException {
        BufferedReader r = new BufferedReader(new InputStreamReader(System.in));
        String notif = r.readLine();
        CentralNotificacoes instance = CentralNotificacoes.getInstance();
        
        while(r.ready()){
            switch(r.readLine()){
                case "nova_instancia": CentralNotificacoes.getInstance(); break;
                case "adicionar": instance.adicionar(notif); break;
            }   
        }
    }
}